### DISCUSSION TITLE
The SPfx package.json is showing " "node": ">=16.13.0 <17.0.0"" while the compatilbitly page is saying (12 | 14 | 16 )

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [pnp/sp-dev-fx-webparts#3568] [Q&A]
I want to deploy the https://github.com/pnp/sp-dev-fx-webparts/tree/main/samples/react-chatgpt-app inside my tenant. but i have this question:-

now inside the web part package.json file it is saying the following regarding node version
`
 "node": ">=16.13.0 <17.0.0"`

while inside the Compatibility section it mentioned the following:-

![image](https://user-images.githubusercontent.com/79726511/223695717-64357c1c-841d-4154-af79-f45635fa259b.png)

so which one we need to use? for example node V12 is valid?

Regards

**Comment by Link631** [comment]
Hi,

well, when you have a look inside package.json you will find out, that following spfx version is being referenced:
1.16.1

For Example following packages are included:
@microsoft/sp-webpart-base": "1.16.1"

Having a look here, you find out, that for spfx 1.16 the node Version v16.13+ is required.
https://learn.microsoft.com/en-us/sharepoint/dev/spfx/compatibility

So node v12 is not valid, you will need a node above v16.13.

Thats why this is described in package.json as followed:
 "node": ">=16.13.0 <17.0.0"

So: Everything between 16.13.x and lower than 17.0.0 but not equals 17.0.0 is a valid node version.

**Reply by mvcsharepointdev** [reply]
@Link631 ok thanks since i opened this question > i used nvm 14.18.1 and i did not get any issues.. so should i be fine.. i faced some errors and i were ale tto fix them by running those commands:-



npm install swiper@6.8.4

npm install adaptivecards adaptivecards-templating adaptive-expressions --save


any advice?

**Reply by Link631** [reply]
Well.... my suggestion is to use the newest npm possible.

But if it runs now in your scenario, it is fine. But be aware that officially it should be node between >=16.13.0 <17.0.0 and that there is no guarantee that it stays fine with your setup.
